source/agora/consensus/EnrollmentManager.d

```diff
@@ -248,16 +248,11 @@ public class EnrollmentManager
         this.enroll_pool.getEnrollments(pool_enrolls, Height(height + 1));
         foreach (enroll; pool_enrolls)
         {
-            const enroll_height = this.getEnrolledHeight(enroll.utxo_key);
-            const avail_height =
```

Comment:
This part of the code should not be removed because it's not related to the current PR. The `avail_height` is to check whether some enrollments can be nominated for the next height or not. Some validators can try to re-enroll earlier before the enrollment expires. For example, a node can be enrolled at the height of 10, and it can try to re-enroll at the height of 11 after being enrolled, and then the `Enrollment` is added to the pool. The added `Enrollment` can be nominated for the height of 1017. That is the original concept for the `avail_height`. If we decide to only use `checkAndEnroll` as the way of re-enrolling, we should remove all the code `avail_height`. Or the code about `avail_height` should remain.

Reply:
With these changes, `avail_height`  is actually completely redundant. This information is implicitly encoded in the `random_seed` since `random_seed` will only be valid for a single height. The validation checks in this function will make sure an enrollment in the pool will only be nominated at the height that it is valid.

We can get rid of `avail_height` in a separate PR I guess.

Reply:
and sending an early Enrollment request may not be desired any more. If you enroll at `Height(0)` with a validator cycle off 1008. You would have to reveal the preimage of `Height(1008)` prematurely just to send an enrollment request for the next cycle.

Reply:
> and sending an early Enrollment request may not be desired any more.

This is the feature we already had, which is done in issue #824 . A validator can re-enroll much earlier before a cycle ends in the situation where `recurring_enrollment` is set to `false`. So the use of `avail_height` could be removed after some discussion and decision.

Reply:
By early I meant sending a request at `Height(1)` for an enrollment that would be available at `Height(1008)`. The behaviour described in #824 is still possible.

Reply:
> By early I meant sending a request at Height(1) for an enrollment that would be available at Height(1008).

this is possible too, just requires the node to reveal a preimage way early.

Reply:
> this is possible too, just requires the node to reveal a preimage way early.

OK, I see what you intended in the code now. Thanks.